### PR TITLE
Define site-wide and library-wide settings in Configuration.

### PR DESCRIPTION
**Pull request by aslagle**

### CODE REVIEW

File: scripts.py [hunk 1]
```diff
@@ -677,7 +690,14 @@ def do_run(self, _db=None, cmd_args=None, output=sys.stdout):
...
                 key, value = setting.split('=', 1)
-                ConfigurationSetting.sitewide(_db, key).value = value
+                if not args.force and not key in [s.get("key") for s in self.config.SITEWIDE_SETTINGS]:
+                    raise ValueError(
+                        "'%s' is not a known site-wide setting. Use --force to set it anyway."
+                        % key
+                    )
+                else:
+                    ConfigurationSetting.sitewide(_db, key).value = value
+                    _db.commit()
```
Comment (leonardr) [inline scripts.py]: Why commit immediately?

Comment (aslagle) [inline scripts.py]: The script didn't seem to be committing at all before...

**Comment by leonardr** [comment]
:+1:

File: scripts.py (hunk 1 above)
Comment (leonardr) [inline scripts.py]: I looked at the code and it looks like `do_run()` is responsible for calling commit. Can you just move this to the end? (and add commit() to any of the other configuration scripts that don't have it at the end of do_run()?)

Comment (aslagle) [inline scripts.py]: Done. The other scripts had it already.

**Pull request merged**